Assets/ZSSRichTextEditor.js

```diff
@@ -273,49 +275,72 @@ ZSSEditor.getJoinedCaretArguments = function() {
     return joinedArguments;
 };
 
+ZSSEditor.getCaretYPosition = function() {
+    var selection = window.getSelection();
+    var range = selection.getRangeAt(0);
+    var span = document.createElement("span");
+    // Ensure span has dimensions and position by
+    // adding a zero-width space character
+    span.appendChild( document.createTextNode("\u200b") );
+    range.insertNode(span);
+    var y = span.offsetTop;
+    var spanParent = span.parentNode;
+    spanParent.removeChild(span);
+    
+    // Glue any broken text nodes back together
+    spanParent.normalize();
+    
+    return y;
+}
+
 ZSSEditor.getYCaretInfo = function() {
-    var y = 0, height = 0;
-    var sel = window.getSelection();
-    if (sel.rangeCount) {
+    var y = 0;
+    var height = 0;
+    var selection = window.getSelection();
+    
+    if (! selection.rangeCount) { // if no selection available just return CaretPositionUnknow
```

Comment:
When comments are used to explain what the code is doing, it usually means that readability can be improved to remove the comment.

For example, instead of:

```javascript

ZSSEditor.getYCaretInfo = function() {
   var y = 0;
   var height = 0;
   var selection = window.getSelection();

   if (! selection.rangeCount) { // if no selection available just return 0
   // ...etc...
}
```

...use...

```javascript
/*
 * @returns {0, 0} if there is no selection.
 */
ZSSEditor.getYCaretInfo = function() {
   var y = 0;
   var height = 0;
   var selection = window.getSelection();

   var noSelection = !selection.rangeCount;

   if (noSelection) {
   // ...etc...
}
```